src/lib/operators/aggregate_hash.cpp

```diff
@@ -632,8 +638,7 @@ std::shared_ptr<const Table> AggregateHash::_on_execute() {
       _aggregate<std::vector<AggregateKeyEntry>>();
       break;
   }
-
-  const auto& input_table = left_input_table();
+  step_performance_data.set_step_runtime(OperatorSteps::Aggregating, timer.lap());
```

Comment:
Separate this into the generation of group IDs and the actual aggregation?

Reply:
Good point. Added a new Step `GroupByKeyPartitioning`.